### PR TITLE
Enhance / refactor some test-suite code

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Split off from https://github.com/bpfkorea/agora/pull/109

### CODE REVIEW

File: source/agora/test/Base.d [hunk 1]
```diff
@@ -303,14 +303,17 @@ public enum NetworkTopology
...
 
     Returns:
         The set of public key added to the node
 
 *******************************************************************************/
 
 public NetworkT makeTestNetwork (NetworkT : TestNetworkManager)
-    (NetworkTopology topology, size_t nodes, bool configure_network = true)
+    (NetworkTopology topology, size_t nodes, bool configure_network = true,
+        long retry_delay = 100, size_t max_retries = 20, long timeout = 500)
```
Comment (Geod24) [inline source/agora/test/Base.d:317]: What are the units of `100` and `500`?

Comment (AndrejMitrovic) [reply source/agora/test/Base.d:317]: I'll document it.

**Comment by Geod24** [comment]
Grey :(

**Comment by AndrejMitrovic** [comment]
Rebased, and added a commit implementing the suggestion at https://github.com/bpfkorea/agora/pull/144#issuecomment-518587806

**Comment by AndrejMitrovic** [comment]
> Grey :(

Green.

**Review by Geod24 — APPROVED** [review]
Re second commit: Looks like we will end up just passing a `Config` struct.

**Pull request merged**